### PR TITLE
adding extension to resize images

### PR DESCRIPTION
**Pull request by Sahil590**

### CODE REVIEW

File: docs/posts/pydata_london_2024.md [hunk 1]
```diff
@@ -49,8 +49,8 @@ Due to the nuance and relative complexity of each application, many ML developer
 
 For streaming and aggregation, attendants were spoilt for choice for low-latency data streaming solutions, with [Bytewax](https://bytewax.io/), [Hopsworks](https://www.hopsworks.ai/) and [CSP](https://docs.cloudera.com/csp-ce/latest/index.html) all presenting solutions. These projects provide reusable building blocks for integrating with various real-time (and offline, for model training) data sources, performing efficient and customisable preprocessing, and presenting to the final data sink as aggregated time-synchronised dataframes.
 
-![Dask documentation](images/pydata_london_2024/dask_document.png)
-
+![Dask documentation](images/pydata_london_2024/dask_document.png){: style="height:400px;width:auto"}
+<!-- markdownlint-disable-next-line MD036 -->
```
Comment (alexdewar) [inline docs/posts/pydata_london_2024.md:53]: I know this is what we've done elsewhere, but I'm wondering whether we should just disable `MD036` in the markdownlint config file.

**Review by alexdewar — COMMENTED** [review]
This seems like the right approach. I did wonder if we could change our CSS file to define a maximum width for images, but that didn't seem to work when I tried it :disappointed:.

Rather than setting the height of each image to be a number of pixels (which will mean the image will be a different size depending on the screen resolution -- in particular it might make things look odd on mobile), I'm wondering if it might be better to set `width` (or `max-width`) to 80% or something.

What does everyone else think?

On a separate note, I found that there is perhaps a cleaner way to display a figure with a caption in markdown: https://squidfunk.github.io/mkdocs-material/reference/images/#image-captions

It does mean dropping into HTML, but I think the result looks nice. Maybe that's something we want to do as well?

**Review by AdrianDAlessandro — CHANGES_REQUESTED** [review]
I like including the `attr_list` extension. I think that can be useful. But I'd like to get to a global option that doesn't require manually setting the size for each figure. I did some digging and found some options:

1. Editing the stylesheets in `docs/stylesheets/extra.css`, and using something like what was suggested in the [discussion you linked](https://github.com/squidfunk/mkdocs-material/discussions/6813). I tried this and it was really difficult to get something sensible. Best I found was this (bu it meant the figure captions didn't align.
```css
.md-content__inner img {
  box-sizing: content-box;
  width: 80%;
  padding-left: 10%;
}
```
2. I then saw [this in the Material theme docs](https://squidfunk.github.io/mkdocs-material/reference/images/#image-captions) about using an extension to render raw html and use the `<figure>` object. But this means _more_ work for each figure.
3. Finally, I found the [MkDocs Caption](https://tobiasah.github.io/mkdocs-caption/) extension that automatically converts `` to `<figure>` and uses the alt-text for the caption. I think this is the best option.

File: docs/posts/pydata_london_2024.md [hunk 2]
```diff
@@ -25,7 +25,7 @@ With Artificial Intelligence, in particular Large Language Models (LLMs) being a
 
 ## Open source software and community building
 
-![Keynote by Tania Allard at PyData London 2024](images/pydata_london_2024/tania_keynote.png)
+![Keynote by Tania Allard at PyData London 2024](images/pydata_london_2024/tania_keynote.png){: style="height:300px;width:auto"}
```
[the remark this answers is not in the document]
Comment (Sahil590) [reply docs/posts/pydata_london_2024.md:28]: Making it global looks good, I did try it earlier but was missing a decency, but now it's all working.

**Comment by Sahil590** [comment]
After adding mkdocs-caption dependency, it's much easier/cleaner to set the default image size.

File: docs/posts/pydata_london_2024.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (Sahil590) [reply docs/posts/pydata_london_2024.md]: I decided to use [markdown_captions](https://github.com/evidlo/markdown_captions) instead

File: mkdocs.yml [hunk 3]
```diff
@@ -28,5 +28,7 @@ plugins:
       categories:
         - categories
         - tags
+
 markdown_extensions:
-  - attr_list
+  - 'markdown_captions'
+  - 'attr_list'
```
Comment (AdrianDAlessandro) [inline mkdocs.yml:34]: Just curious, why are these in 'quotes' now?

File: docs/posts/pydata_london_2024.md [hunk 4]
```diff
@@ -63,25 +57,21 @@ Crafting, calibrating and evaluating models for now-casting and forecasting as w
...
-*Source: Quazi Nafiul Islam’s talk at PyData London 2024*
+![Source: Quazi Nafiul Islam’s talk at PyData London 2024](images/pydata_london_2024/quazi_talk.png)
 
 Particularly exciting to members of the RSE team, Andy Fundinger gave an [overview of the python package 'hypothesis'](https://youtu.be/NL7-eNPr_oI?si=WI7II3v5mt7Wz-b4), a [package](https://hypothesis.readthedocs.io/en/latest/) that allows developers to automatically generate unit tests for their functions, and automates the parameterisation of these tests to cover a wide range of inputs and edge cases. We're excited to implement this tool in current and future projects to increase the robustness of our software.
 
 ## Honourable mentions
 
 ![John Sandall’s talk at PyData London 2024](images/pydata_london_2024/john_talk.png)
-<!-- markdownlint-disable-next-line MD036 -->
 *Source: John Sandall’s talk at PyData London 2024*
```
Comment (AdrianDAlessandro) [inline docs/posts/pydata_london_2024.md:75]:
```suggestion
```

File: docs/stylesheets/extra.css [hunk 5]
```diff
@@ -1,3 +1,7 @@
 :root {
     --md-primary-fg-color:        #0000CD;
   }
+.md-content__inner img {
+  box-sizing: content-box;
+  width: 80%;
```
Comment (AdrianDAlessandro) [inline docs/stylesheets/extra.css:6]: This is a little annoying because they're still all different sizes. But I don't think they look as bad at different sizes when they're figures like this.

Maybe use `max-width` so we don't stretch really small images?
```suggestion
  max-width: 80%;
```

**Review by AdrianDAlessandro — COMMENTED** [review]
What was the reason for using [markdown-captions](https://github.com/evidlo/markdown_captions) instead of [mkdocs-caption](https://github.com/tobiasah/mkdocs-caption)?

The MkDocs one is specifically designed for MkDocs and has more features (like [allowing Alt-text and captions to be different](https://tobiasah.github.io/mkdocs-caption/image/#using-the-title-attribute-as-caption)). It also appears to be actively maintained (last meaningful commit 3 days ago) as opposed to the other one (last meaningful commit 2 years ago)

**Review by AdrianDAlessandro — APPROVED** [review]
Just a couple of comments, but happy with this!
